incubator/webpagetest-server/README.md

```diff
@@ -0,0 +1,213 @@
+# WebPageTest Server
+
+[WebPageTest](https://webpagetest.org/) is a website testing tool
+
+## TL;DR;
+
+```console
+$ helm install incubator/webpagetest-server
+```
+
+## Introduction
+
+This chart bootstraps a private [WebPageTest Server](https://github.com/WPO-Foundation/webpagetest-docs/blob/master/user/Private%20Instances/README.md) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+Depending on your configuration you can then use Official WPT agent instances to test your websites
+
+## Prerequisites
+
+- Kubernetes 1.4+ with Beta APIs enabled
+
+## Installing the Chart
+
+To install the chart with the release name `my-release`:
+
+```console
+$ helm install --name my-release incubator/webpagetest-server
+```
+
+The command deploys WebPageTest on the Kubernetes cluster in the default configuration. The [configuration](#configuration) section lists the parameters that can be configured during installation.
+Please note the default install probably won't do much without some configuration of the values.yaml file
+
+
+> **Tip**: List all releases using `helm list`
+
+## Uninstalling the Chart
+
+To uninstall/delete the `my-release` deployment:
+
+```console
+$ helm delete my-release
+```
+
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following tables lists the configurable parameters of the WordPress chart and their default values.
+
+| Parameter                            | Description                                | Default                                                    |
+| -------------------------------      | -------------------------------            | ---------------------------------------------------------- |
+| `image`                              | WebPageTest server image                   | `webpagetest/server:{VERSION}`                              |
+| `imagePullPolicy`                    | Image pull policy                          | `IfNotPresent`                                             |
+| `ec2Locations.enabled`               | Enables use of EC2 AMI's                   | `false`                                                     |
```

Comment:
Can you fill in details on why ec2 is listed here? Why would a chart be spinning up AMI in one of the public clouds? Would it be better to provide direction to using the agent chart?

Reply:
Webpage test uses various browsers to performance test a website. 
As I'm sure you can appreciate people want to test their site using multiple browsers on different platforms rather than just chrome/firefox on linux.
To the best of my knowledge you cannot have a windows container running on a linux host as such testing with chrome/firefox/ie on windows needs a solution.

Reply:
@timothyclarke First, I appreciate what webpagetest does. I'm a long time user. I remember when it launched. I appreciate the multi-platform capabilities.

That being said, Kubernetes is now adding support for Windows containers. This is a beta feature today. In the future (not expecting this now) you may be able to run windows containers to do that sort of testing.

Reply:
I agree that it is desirable to have this completely within k8s, I do feel strongly that we should not enforce that limit.
The current implementation allows for both a exclusively k8s solution and a hybrid k8s / AMI solution. Granted the current implementations of both the server and agent charts, as well as the wpt agent containers are only possible if you're happy with linux browsers. If you want windows browsers further work is probably needed in the wptagent repo to create a windows container.

My personal belief is that the hybrid solution would be of more benefit to most users even when using windows containers. My reasons are as follows.
- Your k8s cluster is in USA and you would like to perform some tests from Europe or AsiaPac. 
- Different providers would require VPN's etc to extend the k8s cluster over multiple regions / multiple providers.
- Most places simply aren't that big, but may still have those requirements in an adhoc manner.
- It is easier to run an AMI in a remote region than to extend your k8s cluster to that region to perform an adhoc test.